pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -214,3 +234,138 @@ func (s *completed) Execute(msg dispatcher.DIDCommMsg) (state, error) {
 		return nil, fmt.Errorf("illegal msg type %s for state %s", msg.Type, s.Name())
 	}
 }
+
+func sendExchangeRequest(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound, doc *did.Doc) error {
+	requestBytes, dest, err := createExchangeRequest(msg.Payload, doc)
+	if err != nil {
+		return err
+	}
+	//send the exchange request
+	err = outbound.Send(requestBytes, dest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func sendOutbound(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound) error {
```

Comment:
I think we need two functions - `newRequestFromInvitation()` and `sendExchangeRequest()`:

* newRequestFromInvitation() accepts Invitation as parameter as returns a Request and a Destination
* sendExchangeRequest() accepts Request and a Destination and ships it off

Reply:
refactored